### PR TITLE
CI: Performance tests with praktika

### PR DESCRIPTION
**Pull request by maxknv**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [x]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by rschu1ze** [comment]
@maxknv It would be great if you could make the scripts as easily as possible locally runnable. The last time I tried, I failed completely. Ideally, explain in some README the necessary steps to do so. Thanks.

**Comment by maxknv** [comment]
> @maxknv It would be great if you could make the scripts as easily as possible locally runnable. The last time I tried, I failed completely. Ideally, explain in some README the necessary steps to do so. Thanks.

Thanks for letting me know this is needed!
I don't have an option to fail =)

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 399b91de834e9a9c9f312a6219611b2e2bf28496 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/72422/399b91de834e9a9c9f312a6219611b2e2bf28496/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Fast test</td><td>Normally this is the first check that is ran for a PR. It builds ClickHouse and runs most of <a href="https://clickhouse.com/docs/en/development/tests#functional-tests">stateless functional tests</a>, omitting some. If it fails, further checks are not started until it is fixed. Look at the report to see which tests fail, then reproduce the failure locally as described <a href="https://clickhouse.com/docs/en/development/tests#functional-test-locally">here</a></td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/7b48660ef4d5f57e767bcfd4ccee1e74c911bf0e/fast_test.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/c9422a75807666eadcf45271770647dc80063711/unit_tests__release_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: ci/jobs/scripts/perf/compare.sh [hunk 1]
```diff
@@ -569,8 +570,8 @@ unset IFS
 # The comparison script might be bound to one NUMA node for better test
 # stability, and the calculation runs out of memory because of this. Use
 # all nodes.
-numactl --show
-numactl --cpunodebind=all --membind=all numactl --show
+#numactl --show
```
Comment (azat) [inline ci/jobs/scripts/perf/compare.sh:573]: Why this has been commented?

File: ci/jobs/scripts/perf/compare.sh [hunk 2]
```diff
@@ -1434,11 +1441,11 @@ case "$stage" in
     time "$script_dir/report.py" --report=all-queries > all-queries.html 2> >(tee -a report/errors.log 1>&2) ||:
     time "$script_dir/report.py" > report.html
     ;&
-"upload_results")
-    time upload_results ||:
-    ;&
+#"upload_results")
```
Comment (azat) [inline ci/jobs/scripts/perf/compare.sh:1444]: Why this has been commented in the first place?

File: ci/jobs/scripts/perf/compare.sh [hunk 3]
```diff
@@ -224,15 +222,16 @@ function run_tests
...
+# TODO: remove
+#    elif [ "$PR_TO_TEST" -ne 0 ] \
+#        && [ "$(wc -l < changed-test-definitions.txt)" -gt 0 ] \
+#        && [ "$(wc -l < other-changed-files.txt)" -eq 0 ]
+#    then
+#        # If only the perf tests were changed in the PR, we will run only these
+#        # tests. The lists of changed files are prepared in entrypoint.sh because
+#        # it has the repository.
+#        test_files=($(sed "s/tests\/performance/${test_prefix//\//\\/}/" changed-test-definitions.txt))
+#        run_only_changed_tests=1
```
Comment (azat) [inline ci/jobs/scripts/perf/compare.sh:234]: Why this has been commented?

**Comment by maxknv** [comment]
@azat, this PR was my "reasonable effort" to decouple the build job from the perf test job and drop the multilayer structure of performance tests because it makes many things very hard to maintain in CI.

If something that we need does not work after the change it should be fixed.

**Pull request merged**